docs/reviewing-release-prs.md

```diff
@@ -0,0 +1,186 @@
+# Reviewing Release PRs
```

Comment:
Although we’ll likely operate slightly differently from core, with fewer teams contributing, I still think this is a great resource for the release process and something we should keep as we improve at handling releases.